File: vespajlib/src/main/java/com/yahoo/slime/BoolValue.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.slime;

/**
 * @author havardpe
 */
final class BoolValue extends Value {

    private static final BoolValue trueValue = new BoolValue(true);
    private static final BoolValue falseValue = new BoolValue(false);
    private final boolean value;
    private BoolValue(boolean value) { this.value = value; }
    public Type type() { return Type.BOOL; }
    public boolean asBool() { return this.value; }
    public void accept(Visitor v) { v.visitBool(value); }
    public static BoolValue instance(boolean bit) { return (bit ? trueValue : falseValue); }

}

File: client/go/internal/slime/bool_value.go
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
// Author: arnej

package slime

type BoolValue struct {
	valueBase
	v bool
}

func newBoolValue(value bool) *BoolValue {
	return &BoolValue{v: value}
}

func (b *BoolValue) Valid() bool {
	return true
}

func (b *BoolValue) AsBool() bool {
	return b.v
}

func (b *BoolValue) Type() Type {
	return BOOL
}
